src/ui/pages/SplashPage.ts

```diff
@@ -0,0 +1,90 @@
+import { AppConfig } from '../../common/AppConfig';
+import { Assets, Container, Sprite, Text } from 'pixi.js';
+import { Bundles } from '../../App.types';
+import { Events } from '../../common/Events';
+import { FontStyle } from '../../common/Fonts';
+
+/**
+ * The initial page that is displayed until the asset bundles are loaded. Progress information is represented by a preloader.
+ */
+export class SplashPage extends Container {
+    private background: Sprite;
+    private progressLabel: Text;
+
+    /**
+     * 
+     * @param {string} url - URL to manifest
+     */
+    public constructor(url: string) {
+        console.log('SplashPage().constructor()');
+        super();
+
+        this.init(url);
+    }
+
+    /**
+     * Displays the preloader copy immediately and awaits the manifest, then starts loading the assets asyncreniously.
+     * The background is displayed as soon as it's loaded.
+     * 
+     * @param {string} url - URL to manifest
+     */
+    private init = async (url: string) => {
+        console.log('SplashPage().init()');
+
+        this.initPreloader();
+        await this.loadAssets(url);
+        await this.initBackground();
+    }
+
+    private initPreloader = () => {
+        console.log('SplashPage().initPreloader()');
+
+        let settings = AppConfig.getInstance().getSettings();
```

Comment:
it is better to use `const` here because you don't use it again. Also you can spread the settings on the same line like this

Reply:
`const {
width,
height
} = AppConfig.getInstance().getSettings()`

Reply:
nice one! updated as suggested...